addon/macro.js

```diff
@@ -67,37 +68,22 @@ export function raw(value) {
   return new Raw(value);
 }
 
-class TranslationMacro extends ComputedProperty {
+class TranslationMacro extends IntlComputedProperty {
   constructor(translationKey, options) {
     const hash = options || new EmptyObject();
-    const [dynamicValues, staticValues] = partitionDynamicValuesAndStaticValues(hash);
-    const dependentKeys = ['intl.locale'].concat(Object.values(dynamicValues));
-
-    super(function(propertyKey) {
-      let intl = get(this, 'intl');
-
-      assert(
-        `Cannot translate "${translationKey}" as "${propertyKey}".\n${this} does not have an 'intl' property set and there is no 'intl' service registered with the owner.`,
-        intl
-      );
-
-      return intl.t(translationKey, assign({}, staticValues, mapPropertiesByHash(this, dynamicValues)));
-    });
-
-    this.property(...dependentKeys);
-    this.readOnly();
-  }
-
-  setup(proto) {
-    if (super.setup) {
-      super.setup(...arguments);
-    }
-
-    if (!proto.intl) {
-      // Implicitly inject the `intl` service, if it is not already injected.
-      // This allows the computed property to depend on `intl.locale`.
-      defineProperty(proto, 'intl', service('intl'));
-    }
+    const [dynamicValues, staticValues] = partitionDynamicValuesAndStaticValues(
+      hash
+    );
+    const dependentKeys = Object.values(dynamicValues);
+
+    super(
+      (intl, propertyKey, ctx) =>
+        intl.t(
+          translationKey,
+          assign({}, staticValues, mapPropertiesByHash(ctx, dynamicValues))
+        ),
+      ...dependentKeys
+    );
```

Comment:
So considering https://github.com/ember-intl/ember-intl/pull/708/files#r237071761 this could instead also be:

```js
super(...dependentKeys, (intl, propertyKey, ctx) =>
  intl.t(
    translationKey,
    assign({}, staticValues, mapPropertiesByHash(ctx, dynamicValues))
  )
);
```

Which is closer to:

```js
computed(...dependenKeys, getterFn)
```